cmake/sysbuild/partition_manager.cmake

```diff
@@ -0,0 +1,821 @@
+    ${${PM_DOMAIN}${underscore}region_arguments}     # region args are scoped in and thus available. Should probably be in arg.
+    )
+  # ToDo: add_region is done only for main app, how about DOMAIN ?
+
+  set(pm_output_cmd
+    ${PYTHON_EXECUTABLE}
+    ${ZEPHYR_NRF_MODULE_DIR}/scripts/partition_manager_output.py
+    --input-partitions ${pm_out_partition_file}
+    --input-regions ${pm_out_region_file}
+    --config-file ${pm_out_dotconf_file}
+    )
+
+  # Run the partition manager algorithm.
+  execute_process(
+    COMMAND
+    ${pm_cmd}
+    RESULT_VARIABLE ret
+    )
+
+  if(NOT ${ret} EQUAL "0")
+    message(FATAL_ERROR "Partition Manager failed, aborting. Command: ${pm_cmd}")
+  endif()
+
+  # Produce header files and config file.
+  execute_process(
+    COMMAND
+    ${pm_output_cmd}
+    RESULT_VARIABLE ret
+    )
+
+  if(NOT ${ret} EQUAL "0")
+    message(FATAL_ERROR "Partition Manager output generation failed, aborting. Command: ${pm_output_cmd}")
+  endif()
+
+
+  add_custom_target(partition_manager${underscore}${PM_DOMAIN})
+  set(pm_var_names)
+  import_kconfig(PM_ ${pm_out_dotconf_file} pm_var_names)
+
+  foreach(name ${pm_var_names})
+    set_property(
+      TARGET partition_manager${underscore}${PM_DOMAIN}
+      PROPERTY ${name}
+      ${${name}}
+      )
+  endforeach()
+
+  # Turn the space-separated list into a Cmake list.
+  string(REPLACE " " ";" PM_ALL_BY_SIZE ${PM_ALL_BY_SIZE})
+
+  # Iterate over every partition, from smallest to largest.
+  foreach(part ${PM_ALL_BY_SIZE})
+    if(${part} STREQUAL "app")
+      if(DEFINED PM_DOMAIN)
+        set(part "${DOMAIN_APP_${PM_DOMAIN}}")
+      else()
+        set(part "${DEFAULT_IMAGE}")
+      endif()
+    endif()
+    string(TOUPPER ${part} PART)
+    get_property(${part}_PM_HEX_FILE GLOBAL PROPERTY ${part}_PM_HEX_FILE)
+    get_property(${part}_PM_ELF_FILE GLOBAL PROPERTY ${part}_PM_ELF_FILE)
+
+    # Process container partitions (if it has a SPAN list it is a container partition).
+    if(DEFINED PM_${PART}_SPAN)
+      string(REPLACE " " ";" PM_${PART}_SPAN ${PM_${PART}_SPAN})
+      list(APPEND containers ${part})
+    endif()
+
+    # Include the partition in the merge operation if it has a hex file.
+    if(${part} IN_LIST IMAGES)
+      # Question, what is it we want to know ?
+      # We are now in sysbuild, meaning we know everything.
+      # So for each domain we should just include the generated hex file.
+      # Those are available thorugh sysbuild_get, but not locally as there is no parent image.
+      list(APPEND explicitly_assigned ${part})
+      sysbuild_get(${part}_PM_HEX_FILE IMAGE ${part} VAR BYPRODUCT_KERNEL_SIGNED_HEX_NAME CACHE)
+      if(NOT ${part}_PM_HEX_FILE)
+        sysbuild_get(${part}_PM_HEX_FILE IMAGE ${part} VAR BYPRODUCT_KERNEL_HEX_NAME CACHE)
+      endif()
+      set(${part}_PM_TARGET ${part})
+    else()
+      if(${part} IN_LIST images)
+        get_shared(${part}_bin_dir  IMAGE ${part} PROPERTY ZEPHYR_BINARY_DIR)
```

Comment:
`get_shared()` should be replaced with `sysbuild_get()` in this file.

Reply:
code removed, as condition further up prevents this code path from being taken.